src/services/services.ts

```diff
@@ -650,7 +650,7 @@ namespace ts {
         return firstDefined(getAllSuperTypeNodes(classOrInterfaceDeclaration), superTypeNode => {
             const baseType = checker.getTypeAtLocation(superTypeNode);
             const symbol = isStaticMember
-                ? find(checker.getExportsOfModule(baseType.symbol), s => s.escapedName === declaration.symbol.name)
+                ? baseType.symbol && find(checker.getExportsOfModule(baseType.symbol), s => s.escapedName === declaration.symbol.name)

```

Comment:
To quote https://github.com/microsoft/TypeScript/pull/46719#discussion_r869821627

> `getExportsOfModule` on the non-merged symbol of something that isn't a module? This doesn't seem right.

Can another reviewer verify that this was not the right code from the get-go?

Reply:
I guess `.symbol` is fine when fetched off of a type; everything should be resolved by that point.

Reply:
`getAllSuperTypeNodes` returns all _super_ nodes, even if some of them are invalid. 

```ts
var A: any;
class B extends A {}
```

Should `getTypeAtLocation(/*A*/)` return type with symbol?

Do we need to avoid checking symbols in the following cases?

https://github.com/microsoft/TypeScript/blob/56a4a93718ce016f038bcbce425bd3f47c3bec78/src/services/findAllReferences.ts#L2273-L2278

https://github.com/microsoft/TypeScript/blob/d879880a3778d02b57642251e0d3d719ec95599a/src/services/completions.ts#L3272-L3277

Reply:
> Should `getTypeAtLocation(/*A*/)` return type with symbol?

The comment here

https://github.com/microsoft/TypeScript/blob/1ba609660be56189165896ce09f9902b7b9ae96d/src/compiler/types.ts#L5266

implies that the `.symbol` might not be set, which probably has to do with the `errorType`, or a type name that doesn't actually exist.

Checking if `someType.symbol` is present in those cases is likely fine. You can see whether any tests fail if it's not present.

Let me know if that answers the questions sufficiently

Reply:
I'm pretty sure `getExportsOfModule` should actually be `getExportsOfSymbol`, although that's not currently exported. 

*However*, `getExportsOfSymbol` ends up returning `symbol.exports` after checking for a bunch of other things that won't be present on a class, like `default`, `export=` and `export *`. So in practise it's probably not going to crash.

I think we should expose getExportsOfSymbol publically, though. It wraps getExportsOfModule so there's no good reason to expose the latter without exposing the former.